flink-contrib/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackend.java

```diff
@@ -222,11 +227,40 @@ public CheckpointStreamFactory createStreamFactory(JobID jobId,
 
 	@Override
 	public CheckpointStreamFactory createSavepointStreamFactory(
-			JobID jobId,
 			String operatorIdentifier,
 			String targetLocation) throws IOException {
 
-		return checkpointStreamBackend.createSavepointStreamFactory(jobId, operatorIdentifier, targetLocation);
+		return checkpointStreamBackend.createSavepointStreamFactory(operatorIdentifier, targetLocation);
+	}
+
+	@Override
+	public boolean supportsExternalizedMetadata() {
+		return checkpointStreamBackend.supportsExternalizedMetadata();
+	}
+
+	@Nullable
+	@Override
+	public String getMetadataPersistenceLocation() {
```

Comment:
The  meta data location might be a concept that justifies introducing a class. In particular, it seems that there might be different location concepts in the future, so simply encoding all this in a string is questionable.